final/index.html

```diff
@@ -0,0 +1,180 @@
+        width: 200px;
+        height: 200px;
+      }
+
+      #update-time {
+        margin-top: 8px;
+      }
+    </style>
+  </head>
+
+  <body class="fit vertical layout center-center">
+    <template is="dom-bind" id="t">
+      <platinum-sw-register skip-waiting
+                            clients-claim
+                            auto-register>
+        <platinum-sw-import-script href="weather-fetch-handler.js"></platinum-sw-import-script>
+        <platinum-sw-fetch handler="weatherFetchHandler"
+                           path="/v1/public/yql"
+                           origin="https://query.yahooapis.com">
+        </platinum-sw-fetch>
+        <platinum-sw-cache default-cache-strategy="fastest"></platinum-sw-cache>
+      </platinum-sw-register>
+
+      <geo-location latitude="{{latitude}}"
+                    longitude="{{longitude}}"
+                    watch-pos
+                    on-geo-error="_requestWeatherData"
+                    on-geo-response="_requestWeatherData">
+      </geo-location>
+
+      <iron-ajax id="weather-ajax"
+                 url="[[weatherApiUrl]]"
+                 params="[[_calculateWeatherParams(latitude, longitude)]]"
+                 handle-as="json"
+                 on-response="_handleWeatherResponse">
+      </iron-ajax>
+
+      <template is="dom-if" if="[[isLoading]]">
+        <h3>Loading...</h3>
+        <paper-spinner active></paper-spinner>
+      </template>
+
+      <template is="dom-if" if="[[weatherResponse]]">
+        <div class="vertical center layout">
+          <h3>
+            Weather for
+            <span>[[_calculateLocationString(weatherResponse.query.results.channel.location)]]</span>
+          </h3>
+          <iron-icon id="weather-icon"
+                     icon="[[_calculateWeatherIcon(weatherResponse.query.results.channel.item.condition.code)]]">
+          </iron-icon>
+          <h2>
+            <span>[[weatherResponse.query.results.channel.item.condition.temp]]</span>°
+            <span>[[weatherResponse.query.results.channel.units.temperature]]</span>,
+            <span>[[weatherResponse.query.results.channel.item.condition.text]]</span>
+          </h2>
+          <div class="horizontal justified layout">
+            <h5 id="update-time">
+              As of <span>[[weatherResponse.query.results.channel.item.condition.date]]</span>
+            </h5>
+            <paper-icon-button icon="refresh"
+                               disabled$="[[disableRefreshButton]]"
+                               on-click="_requestWeatherData">
+            </paper-icon-button>
+          </div>
+          <a href="https://www.yahoo.com/?ilc=401" target="_blank">
+            <img src="https://poweredby.yahoo.com/purple.png">
+          </a>
+        </div>
+      </template>
+    </template>
+
+    <script>
+      var t = document.querySelector('#t');
+      t.weatherApiUrl = 'https://query.yahooapis.com/v1/public/yql';
+      t.latitude = 40.7413065;
```

Comment:
They're fallbacks in case someone denies Location permission, so the app at least shows something. I'll make that clearer with a comment.